lib/Resizable.js

```diff
@@ -4,227 +4,492 @@ import PropTypes from 'prop-types';
+
+    resizeFromBottomLeftCorner = (x, y, canDragY, deltaY, deltaX) => {
+        let dimension = this.getCurrentDimension()
+        const currentLeft = this.getComponentLeftPosition()
+        dimension.height = this.state.height + (canDragY ? deltaY : 0);
+
+        return this.isMouseMovingLeft(deltaX) ?
+            this.increaseWidthMoveLeft(currentLeft, x, dimension,deltaX) :
+            this.reduceWidthMoveRight(currentLeft, x, dimension,deltaX)
+    }
+
+    resizeFromBottomRightCorner = (deltaY, deltaX, canDragY, canDragX) => {
+        let dimension = this.getCurrentDimension()
+        dimension.height = this.state.height + (canDragY ? deltaY : 0);
+        dimension.width = this.state.width + (canDragX ? deltaX : 0);
+        return dimension
+    }
+
+    isResizeHandle = (handleName) => {
+        if (handleName) {
+            return handleName.indexOf('-resizable-handle') != -1 && handleName.indexOf('react-') != -1
+        } else {
+            return false
+        }
+    }
+
+    getCurrentDimension = () => (
+        {
+            width: this.state.width,
+            height: this.state.height,
+            left: this.state.left,
+            top: this.state.top
+        }
+    )
+
+    getDimensionAcordingToMousePosition = (handleName, x, y, canDragX, canDragY, deltaX, deltaY) => {
+        switch (handleName) {
+            case 'react-nw-resizable-handle':
+                return this.resizeFromTopLeftCorner(x, y, deltaX, deltaY)
+            case 'react-ne-resizable-handle':
+                return this.resizeFromTopRightCorner(x, y, canDragX, deltaX,deltaY)
+            case 'react-sw-resizable-handle':
+                return this.resizeFromBottomLeftCorner(x, y, canDragY, deltaY,deltaX)
+            case 'react-resizable-handle':
+            case 'react-se-resizable-handle':
+                return this.resizeFromBottomRightCorner(deltaY, deltaX, canDragY, canDragX)
+            default :
+                return null
+        }
+    }
+
+    onResize = (event, {node, deltaX, deltaY, x, y}) => {
+        const canDragX = this.props.axis === 'both' || this.props.axis === 'x';
+        const canDragY = this.props.axis === 'both' || this.props.axis === 'y';
+
+        let width = this.state.width
+        let height = this.state.height
+        let left = this.state.left
+        let top = this.state.top
+
+        if (this.isResizeHandle(this.state.resizableHandle)) {
+            const dimension = this.getDimensionAcordingToMousePosition(this.state.resizableHandle,
+                x, y, canDragX, canDragY, deltaX, deltaY)
+
+            width = dimension.width
+            height = dimension.height
+            top = dimension.top
+            left = dimension.left
+        }
+
+        [width, height] = this.runConstraints(width, height)
+
+        if (!this.hasDimensionChanged(width, height, left, top)) {
+            return;
+        }
+
+        const resizeState = this.onResizeState(width,height,left,top)
```

Comment:
I see no spaces, please reformat the file

Reply:
done